Add MUI2 support for Cover Tabs

YannickMG opened the pull request:
Only converted Covers will display properly.

This should currently be visible to players only when MUI2 is force-enabled.

### Cover converted to MUI2
![image](https://github.com/user-attachments/assets/83bb5485-5457-47af-a439-b170b85108e9)

### Unconverted cover
![image](https://github.com/user-attachments/assets/53011703-39e4-4fc6-8cf9-9988f95a4413)

miozune commented on src/main/java/gregtech/common/covers/gui/CoverGui.java:
```diff
@@ -62,11 +79,9 @@ public ModularPanel createBasePanel(CoverGuiData guiData, PanelSyncManager syncM
-        // ButtonWidget.closeWindowButton(true)
-        // .setPos(getGUIWidth() - 15, 3));
-        // }
+        if (guiData.isAnotherWindow()) {
```
I'm not sure, but what do you think about removing `isAnotherWindow` from `CoverGuiData`? Due to data flow both sides should already be knowing whether it's on another window without syncing. Instead we can simply pass `isAnotherWindow` to `createBasePanel`.

miozune commented on src/main/java/gregtech/common/covers/Cover.java:
```diff
@@ -294,7 +294,19 @@ protected GTGuiTheme getUITheme() {
+     * @param syncManager sync handler where widget sync handlers should be registered
+     * @return UI panel to show
+     */
+    public final ModularPanel buildPopUpUi(CoverGuiData guiData, String panelName, PanelSyncManager syncManager) {
```
nit: Even though we often use `Gui` rather than `GUI`, `Ui` looks odd especially when it's next to `buildUI`.

YannickMG commented:
I went a little bit further than requested. After removing CoverGuiData.isAnotherWindow it turns out we really didn't need any additional data from CoverGuiData since it is intended to be linked to only a single cover instance and we have made sure that this instance is represented by a single object since the Cover refactor. Since we're also creating new instances of CoverGui on demand, it's not much additional cost to give those a reference to the cover they are rendering. As a bonus we can stop casting the results of `CoverGuiData::getCover` to the  right subclass and use a properly typed field to hold the Cover instance.

After we backport upstream features that automatically close the UI when it is invalidated, there will be no downside to holding on to a reference of the cover.

miozune commented:
Thank you, the way cover GUIs are organized is exactly what I've wanted them to be!
I think we can remove `CoverGuiData` now.

YannickMG commented:
> Thank you, the way cover GUIs are organized is exactly what I've wanted them to be! I think we can remove `CoverGuiData` now.

I think we need to keep `CoverGuiData`. Even though we hadn't been doing it previously, we should validate whether the cover on the clientside matches the cover for which the UI is opened on the server side. Otherwise we will get mistmatched packets and either a crash or data corruption. I added a check which crashes the client anyway in such cases but with a more informative message.

I also implemented `UIFactory::canInteractWith` so we can close the GUI any time the cover becomes invalid or should otherwise be inaccessible to the player, such as them being out or interaction range.

YannickMG commented:
Requesting a review from all developers in case miozune doesn't have time to get to it, as they have signaled that they might not have time for MUI2 at present.
